src/RL/representation/default/cp_layer/types.jl

```diff
@@ -81,7 +81,7 @@ struct CPLayerGraph <: LightGraphs.AbstractGraph{Int}
     """
     function CPLayerGraph(cpmodel::CPModel)
         variables = Set{AbstractVar}(values(cpmodel.variables))
-        valuesOfVariables = branchable_values(cpmodel)
+        valuesOfVariables = sort(branchable_values(cpmodel))

```

Comment:
can you explain briefly what the issue was with the previous pipeline ?

Reply:
In short, the ordering of the actions taken by the agent was not consistent with the ordering of value node features in the graph. As a result, the values embeddings in `FullFeaturedCPNN` didn't match the actions taken by the agent afterwards. But this fix doesn't impact other CPNNs.